lib/model/model.go

```diff
@@ -955,6 +1004,25 @@ func (m *Model) IsPaused(device protocol.DeviceID) bool {
 	return paused
 }
 
+func (m *Model) TemporaryIndexSubscribers() map[string][]Connection {
```

Comment:
This thing worries me slightly, and the place in progressEmitter where it's called. It's a large double looped thing creating rather heavy structures and probably doing lots of allocations, called fairly frequently and then looped over. Is there some way we can precompute or simplify this so it's a little less work and allocations done on every announce interval?